src/config.jl

```diff
@@ -20,11 +20,46 @@ end
 
 Tag(::Nothing, ::Type{V}) where {V} = nothing
 
-
 @inline function ≺(::Type{Tag{F1,V1}}, ::Type{Tag{F2,V2}}) where {F1,V1,F2,V2}
     tagcount(Tag{F1,V1}) < tagcount(Tag{F2,V2})
 end
 
+# SmallTag is similar to a Tag, but carries just a small UInt64 hash, instead
+# of the full type, which makes stacktraces / types easier to read while still
+# providing good resilience to perturbation confusion.
+struct SmallTag{H}
+end
```

Comment:
This should be an actual docstring

Reply:
And it should be marked public or exported, but so should other symbols (#744)

Reply:
Are we sure this needs a new type at all? Could we just extend `Dual`'s first type parameter from `Union{Nothing, Tag}` to perhaps `Union{Nothing, Tag, Integer}`?
```
julia> ForwardDiff.Dual(pi/2, 1)
Dual{Nothing}(1.5707963267948966,1.0)

julia> ForwardDiff.derivative(x -> @show(x), pi/2)
x = Dual{ForwardDiff.Tag{var"#37#38", Float64}}(1.5707963267948966,1.0)
1.0
```

Reply:
If it does need a new type, the obvious name is `HashTag`.

Reply:
IMO it should be a type. Otherwise people might start passing 1 by accident. A UInt might be ok though

Reply:
Yes it's safer as a type, so it should be a type.